### ISSUE TITLE
Falcon Demo Debug

### ISSUE THREAD

**Issue by ashayestehmanesh** [tenstorrent-metal/tt-metal#2549]
1. Remove the padding from CPU demo, and make GS PREFILL stage to match
- Use the attention mask on CPU to also mask out the tokens used as paddings; the output should be the same if the pad tokens didn't exists! This is CPU vs CPU test! (one with padding+mask out vs no padding)
- to get GS replicate the with padding+mask out version

2. Make the decode stage of GS to match CPU
- Let's make a test where PREFILL is done on CPU; so that the KV-cache and the first token fed to decode match fully
- Debugging!

3. Gradually bring back the model to use tt-ops
- replace torch matmuls to tt matmuls
- softmax
- make sure the model is fully on TT
